### ISSUE TITLE
JAVA_HOME, asdf and jdtls

### ISSUE THREAD

**Issue by Cletip** [emacs-lsp/lsp-java#381]
Hello,

I am with ubuntu and I have installed, with asdf, openjdk version "16.0.2" 2021-07-20.

Moreover, when I do the command 
echo $JAVA_HOME
I have the following result: 
/home/user/.dotfiles/.asdf/installs/java/openjdk-16.0.2
Which is for me quite normal. 

Moreover, I can compile and launch a java program.

However, when I try to install jdtls with the following command: lsp-install-server jdtls, I get this error: 
JAVA_HOME is not defined correctly.

Complete error: 
sh mvnw -Djdt.js.server.root\=/home/user/.dotfiles/.emacs.d/var/lsp-java/eclipse.jdt.ls/server/ -Djunit.runner. root\=/home/user/.dotfiles/.emacs.d/var/lsp-java/eclipse.jdt.ls/test-runner/ -Djunit.runner.fileName\=junit-platform-console-standalone. jar -Djava.debug.root\=/home/user/.dotfiles/.emacs.d/var/lsp-java/eclipse.jdt.ls/server/bundles clean package -Djdt.download.url\=https\://download.eclipse.org/jdtls/milestones/1.5.0/jdt-language-server-1.5.0-202110191539.tar.gz
Error: JAVA_HOME is not defined correctly.
  We cannot execute 

Thanks in advance for your future answers

**Comment by yyoncho** [comment]
maybe you need https://github.com/purcell/exec-path-from-shell

**Comment by Cletip** [comment]
That's works perfectly ! Thanks you a lot, really ! I post the solution here : 

(use-package exec-path-from-shell
      :config
      (exec-path-from-shell-initialize)
      )

**Comment by Cletip** [comment]
That's works perfectly ! Thanks you a lot, really ! I post the solution here : 

(use-package exec-path-from-shell
:config
(exec-path-from-shell-initialize)
)

**Comment by max-nazarati** [comment]
Had the same issue on MacOS with zsh and homebrew-installed Emacs. What worked for me is moving the `export JAVA_HOME...` to the `~/.zshenv` file instead of the `~/.zshrc`. Apparently the latter one is intended for configuring an interactive shell and Emacs does not make use of it.

**Comment by Stratus3D** [comment]
Hi everyone, sorry to post on an old issue here. I'm one of the maintainers of asdf and I had trouble installing jdtls properly on my system and ended up creating https://github.com/Stratus3D/asdf-jdtls. I figured I'd share in case anyone else comes across this thread and doesn't want to install jdtls manually and already has asdf installed. The only caveat with the asdf plugin right now is that it doesn't take into account the Java version when installing jdtls. There is a thread on this here: https://github.com/eclipse-jdtls/eclipse.jdt.ls/discussions/3597 I had to revert to trial and error when selecting the right jdtls version for my java projects.
